### ISSUE TITLE
Improve glTF plugin compatibility with newer ScriptedImporterAttribute syntax

### ISSUE THREAD

**Issue by bitinn** [atteneder/glTFast#352]
**Is your feature request related to a problem? Please describe.**

I am testing interoperability across various glTF plugins for Unity, and run into an issue between multiple ScriptedImporter declaring itself as default importer for `glb` or `gltf` file type.

Unity will disable all importers for `glb` and `gltf` files upon seeing this, and [an error will be logged](https://github.com/Unity-Technologies/UnityCsReference/blob/master/Modules/AssetPipelineEditor/Public/ScriptedImporter.cs#L116):

> Multiple scripted importers are targeting the extension 'glb' and have all been rejected: ...

**Describe the solution you'd like**

[This fork of UnityGLTF](https://github.com/prefrontalcortex/UnityGLTF) is the only plugin that doesn't conflict with other glTF importer via the use of `overrideExts` in ScriptedImporterAttribute (an added feature since 2020.2).

https://docs.unity3d.com/2022.2/Documentation/ScriptReference/AssetImporters.ScriptedImporterAttribute.html
https://github.com/prefrontalcortex/UnityGLTF/blob/master/UnityGLTF/Assets/UnityGLTF/Editor/Scripts/GLTFImporter.cs#L24

**Describe alternatives you've considered**

N/a

**Additional context**

I have tested this change to work:

![Screenshot 2022-03-04 140543](https://user-images.githubusercontent.com/1484279/156710978-ff3e9138-3c63-4029-a66a-6d9504d99c20.jpg)

**Comment by bitinn** [comment]
Bad news: if no plugin declare `exts` for a file extension (eg. they all declare `overrideExts` only), then under Unity 2022.2.x at least, no Inspector will show up; but if there is more than 1 plugin declare `exts` for a file extension, then Unity disable all of them...

I will follow up with Unity soon.

https://github.com/prefrontalcortex/UnityGLTF/issues/37#issuecomment-1058886899

**Comment by atteneder** [comment]
@bitinn Thanks for this excellent report! I haven't looked into 2022.1 too much (especially with multiple, conflicting importers), but I'll do so next week.

**Comment by bitinn** [comment]
Ah my unity doc link was misleading, I was using 2021.2.10f1 to test all of these. The new attribute syntax is available since 2020.2; the conflict / no importer problem probably exists for all versions.

**Comment by atteneder** [comment]
I tried what you suggested only to find out that the import won't work anymore if you have just one package installed (e.g. glTFast only or UnityGLTF only).

I'm going to ask around how this should be handled. Maybe @hybridherbst knows more.

**Comment by bitinn** [comment]
@atteneder yes I am aware of this issue, which I did mention in:

1. https://github.com/prefrontalcortex/UnityGLTF/issues/37#issuecomment-1058886899
2. https://forum.unity.com/threads/scriptedimporterattribute-handling-blocks-multiple-importers-for-the-same-file-type.1248427/

I hope you can ping Unity team to look into it :)

**Comment by hybridherbst** [comment]
@bitinn thanks for opening the issue on both ends! I'm also aware of this issue, and had opted to use the overrideExt in UnityGltf since we're often using that along with glTFast. I agree that's not ideal.

I have in fact talked to the relevant Unity team about it, and they're aware but don't have a good solution. As so often, this override feature was designed for single projects, not for independent packages. The goal was to have clear guarantees on which importer is used, so there's an error if two say they're "default" and only the override is stored if needed.

@atteneder I've sent you the relevant link, maybe we can both push for a fix to this (for example, adding a "priority" value would fix this).

My understanding is that the only workaround for now (but one we could easily implement actually) is to ifdef the ext either on or off with a global scripting define per package.
This could theoretically also be implicit based on existence of known glTF import packages (if there's an agreed "hierarchy" between them.

I'll go ahead and update UnityGltf for this - it's actually one of the last issues before finally making a PR back to the main repo...

**Comment by hybridherbst** [comment]
OK, UnityGLTF will now check for glTFast existance, and use the regular scripted importer if it doesn't exist and overrideExt if it does exist.
Additionally, I added the ability to override via global scripting defines. I made a PR for glTFast as well which will, by default, not change anything but allow to specify the default importer via global define as well: 
https://github.com/atteneder/glTFast/pull/367

If this PR doesn't get in it's not _that_ bad (by default should still work at least with UnityGltf / other known importers that respect glTFast being the default), it would just allow for explicit control.
